File: farm/farm-front/src/app/services/producao/producao.service.ts
import { Injectable } from '@angular/core';
import {BaseService} from '../base.service';
import {HttpClient} from '@angular/common/http';
import {API_PATH} from '../../app.const';

@Injectable()
export class ProducaoService extends BaseService {

  constructor(http: HttpClient) {
    super(http, '/api/producao');
  }

  findByTalhaoId(page: number, count: number, talhaoId: string) {
    return this.http.get(`${API_PATH}/${this.path}/${page}/${count}/${talhaoId}`);
  }

}

File: farm/farm/src/main/java/com/glauco/farm/api/services/producao/ProducaoService.java
package com.glauco.farm.api.services.producao;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import com.glauco.farm.api.model.Producao;
import com.glauco.farm.api.services.BaseService;

@Component
public interface ProducaoService extends BaseService<Producao> {

    Page<Producao> findByTalhaoId(int page, int count, String talhaoId);

    Producao findByTalhaoIdAno(String id, Integer ano);

}
